main/src/ca/uwaterloo/flix/api/lsp/provider/completion/WithHandlerCompleter.scala

```diff
@@ -0,0 +1,25 @@
+package ca.uwaterloo.flix.api.lsp.provider.completion
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.api.lsp.{Index, TextEdit}
+import ca.uwaterloo.flix.language.ast.TypedAst
+
+object WithHandlerCompleter {
+  def getCompletions(context: CompletionContext)(implicit flix: Flix, index: Index, root: TypedAst.Root): Iterable[Completion] = {
+    if (context.previousWord != "with") {
+      return Nil
+    }
+
+    // TODO: infer the effect of the expression
```

Comment:
What this about?

Reply:
Ideally when writing 
```flix
try f() with 
```
you would only get the snippet for handling the effect that `f` has, if it was known. I couldn't get that to work unfortunately, and I don't know if we even have that information when the full statement isn't parsable, so now it's just suggesting all effects.

Reply:
I see. Thats going to be difficult. I would just focus on offering a list of suggestions once a letter has been typed.